src/segment-loader.js

```diff
@@ -2325,41 +2308,10 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     if (!this.hasEnoughInfoToLoad_()) {
       this.loadQueue_.push(() => {
-        const buffered = this.buffered_();
-
-        if (typeof segmentInfo.timestampOffset === 'number') {
-          // The timestamp offset needs to be regenerated, as the buffer most likely
-          // changed since the function was added to the queue. This is expected, as the
-          // load is usually pending the main loader appending new segments.
-          //
-          // Note also that the overrideCheck property is set to true. This is because
-          // isPendingTimestampOffset is set back to false after the first set of the
-          // timestamp offset (before it was added to the queue). But the presence of
-          // timestamp offset as a property of segmentInfo serves as enough evidence that
-          // it should be regenerated.
-          segmentInfo.timestampOffset = timestampOffsetForSegment({
-            segmentTimeline: segmentInfo.timeline,
-            currentTimeline: this.currentTimeline_,
-            startOfSegment: segmentInfo.startOfSegment,
-            buffered,
-            overrideCheck: true
-          });
-        }
-
-        delete segmentInfo.audioAppendStart;
-
-        const audioBuffered = this.sourceUpdater_.audioBuffered();
-
-        if (audioBuffered.length) {
-          // Because the audio timestamp offset may have been changed by the main loader,
-          // the audioAppendStart should be regenerated.
-          //
-          // Since the transmuxer is using the actual timing values, but the buffer is
-          // adjusted by the timestamp offset, the value must be adjusted.
-          segmentInfo.audioAppendStart = audioBuffered.end(audioBuffered.length - 1) -
-            this.sourceUpdater_.audioTimestampOffset();
-        }
-
+        // regenerate the audioAppendStart, timestampOffset, etc as they
+        // may have changed since this function was added to the queue.
+        this.isPendingTimestampOffset_ = true;
```

Comment:
to force timestampoffset to regenerate in generateSegmentInfo.

Reply:
It feels a bit strange to set the property here and then clear it as a side effect of `generateSegmentInfo` (we definitely had side effects before, but in the interest of keeping the new function as side effect free as possible). I wonder if it would be better to set `isPendingTimestampOffset_` to `false` in `updateTransmuxerAndRequestSegment_` and then to have another option that can be passed into `generateSegmentInfo_` to regenerate relevant timing info.

Reply:
This was a bit more complicated than I thought, but I generally agreed with your points here. There were two considerations: `vtt-segment-loader` never wants to set timestamp offsets on generated segment info. I got around that by making a member function that vtt-segment-loader will inherit and override to return null.

Second was that whenever `timestampOffset` is set using the `timestampOffsetForSegment` function we expect to set `isPendingTimestampOffset_` to `false`. So we will still have a small side effect no matter what we do, as our code currently expects us to do that. That being said I did add an option to force a timestamp update so that we don't have to set `isPendingTimestampOffset_` where the original comment was.